pkg/registries/registries.go

```diff
@@ -0,0 +1,93 @@
+package registries
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/unshare"
+	"github.com/docker/distribution/reference"
+	"github.com/pkg/errors"
+)
+
+// userRegistriesFile is the path to the per user registry configuration file.
+var userRegistriesFile = filepath.Join(os.Getenv("HOME"), ".config/containers/registries.conf")
+
+// SystemRegistriesConfPath returns an appropriate value for types.SystemContext.SystemRegistriesConfPath
+// (possibly "", which is not an error), taking into account rootless mode and environment variable overrides.
+//
+// FIXME: This should be centralized in a global SystemContext initializer inherited throughout the code,
+// not haphazardly called throughout the way it is being called now.
+func SystemRegistriesConfPath() string {
```

Comment:
Overall, just don’t add this subpackage at all, please.

E.g. “the code” and “haphazardly called throughout” applies to the original package, not (yet?) to c/common.

In general, designing stable library APIs requires notably more thought (and clairvoyance about the future) than designing private interfaces that can change any time; so let’s not add any that we don’t need to (and when we do add some, I lean towards being extremely specific about the purpose of the API instead of providing a general “read all internal data” accessors and letting callers assign some unknown/unpredictable/hard-to-control. semantics to it” approach; so, if we did have to add an API, I’d prefer something like `DefaultLoginRegistryWhenUnspecified`. Adding more APIs is almost free, changing existing functions in released libraries is comparatively very costly.)